docs/SOAP-Service-Methods.md

```diff
@@ -582,6 +584,66 @@ Unblock activation with given ID. Activations can be unblocked in BLOCKED state
 | `String` | `activationId` | An identifier of an activation |
 | `ActivationStatus` | `activationStatus` | An activation status |
 
+### Method 'lookupActivations'
+
+Lookup activations using query parameters.
+
+#### Request
+
+`LookupActivationsRequest`
+
+| Type | Name | Description |
+|------|------|-------------|
+| `String` | `userIds` | User IDs to use in query, at least one user ID needs to be specified |
+| `String` | `applicationIds` | Application IDs to use in the query, do not specify value for all applications |
+| `String` | `timestampLastUsedBefore` | Filter activations by timestamp when the activation was last used (timestampLastUsed < timestampLastUsedBefore), do not specify value for any timestamp |
+| `String` | `timestampLastUsedAfter` | Filter activations by timestamp when the activation was last used (timestampLastUsed >= timestampLastUsedAfter), do not specify value for any timestamp |
+| `String` | `activationStatus` | Filter activations by their status, do not specify value for any status |
+
+#### Response
+
+`LookupActivationsResponse`
+
+| `Activation[]` | `activations` | A collection of activations for given query parameters |
+
+`LookupActivationsResponse.Activation`
+
+| Type | Name | Description |
+|------|------|-------------|
+| `String` | `activationId` | An identifier of an activation |
+| `ActivationStatus` | `activationStatus` | An activation status |
+| `String` | `blockedReason` | Reason why activation was blocked (default: NOT_SPECIFIED) |
+| `String` | `activationName` | An activation name |
+| `String` | `extras` | Any custom attributes |
+| `DateTime` | `timestampCreated` | A timestamp when the activation was created |
+| `DateTime` | `timestampLastUsed` | A timestamp when the activation was last used |
+| `DateTime` | `timestampLastChange` | A timestamp of last activation status change |
+| `String` | `userId` | An identifier of a user |
+| `Long` | `applicationId` | An identifier fo an application |
+| `String` | `applicationName` | An application name |
+| `Long` | `version` | Activation version |
+
+### Method 'updateStatusForActivations'
+
+Update status for activations identified using their identifiers.
+
+#### Request
+
+`UpdateStatusForActivationsRequest`
+
+| Type | Name | Description |
+|------|------|-------------|
+| `String[]` | `activationIds` | Identifiers of activations whose status needs to be updated |
+| `ActivationStatus` | `activationStatus` | Activation status to use when updating the activations |
+
+#### Response
+
+`UpdateStatusForActivationsResponse`
+
+| Type | Name | Description |
+|------|------|-------------|
+| `boolean` | `updated` | Whether status update succeeded for given activations |
```

Comment:
If possible, we can also return a bit more info here. Something in the sense:

| Type | Name | Description |
|------|------|-------------|
| `boolean` | `updated` | Whether status update succeeded for <del>given</del> **all provided** activations |
| `String[]` | `updateFailedActivations` | In case update failed for some activations, this array contains a list of activations where the update failed. |

Looking at the current implementation, we always return `true`, though... If we are not able to provide more data here, please ignore this comment...

Reply:
As far as I know, the whole operation either succeeds or fails - in case any SQL query fails all updates are rolled back. I only included the `updated` boolean so that a non-empty response is returned. Indeed, it always returns true. In case of any SQL error an error response is returned and the whole transaction is rolled back.